velox/exec/benchmarks/PrefixSortBenchmark.cpp

```diff
@@ -0,0 +1,224 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "glog/logging.h"
+#include "velox/exec/PrefixSort.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace {
+
+// Init memoryManager
+memory::MemoryPool* rootPool() {
+  static std::shared_ptr<memory::MemoryPool> rootPool_ = []() {
+    memory::MemoryManager::initialize({});
+    return memory::memoryManager()->addRootPool("PrefixSortTest", 4L << 30);
+  }();
+  return rootPool_.get();
+}
+
+class TestCase {
+ public:
+  TestCase(
+      memory::MemoryPool* pool,
+      const std::string& testName,
+      size_t rowsNum,
+      const RowTypePtr& rowType)
+      : testName_(testName), rowsNum_(rowsNum) {
+    data_ = std::make_unique<RowContainer>(rowType->children(), pool);
+    RowVectorPtr sortedRows = makeFuzzRows(pool, rowsNum, rowType);
+    storeRows(rowsNum, sortedRows, data_.get(), rows_);
+  };
+
+  const std::string& testName() const {
+    return testName_;
+  }
+
+  const size_t rowsNum() const {
+    return rowsNum_;
+  }
+
+  const std::vector<char*>& rows() const {
+    return rows_;
+  }
+
+  RowContainer* rowContainer() const {
+    return data_.get();
+  }
+
+ private:
+  // Store sortedRows into a RowContainer data to mock the behavior of
+  // SortBuffer.
+  void storeRows(
+      const int rowsNum,
+      const RowVectorPtr& sortedRows,
+      RowContainer* data,
+      std::vector<char*>& rows) {
+    SelectivityVector allRows(rowsNum);
+    rows.resize(rowsNum);
+    for (int row = 0; row < rowsNum; ++row) {
+      rows[row] = data->newRow();
+    }
+    for (int column = 0; column < sortedRows->childrenSize(); ++column) {
+      DecodedVector decoded(*sortedRows->childAt(column), allRows);
+      for (int i = 0; i < rowsNum; ++i) {
+        char* row = rows[i];
+        data->store(decoded, i, row, column);
+      }
+    }
+  }
+
+  // Fuzz test rows : for front columns (column 0 to size -2) use high
+  // nullRatio to enforce all columns to be compared.
+  RowVectorPtr makeFuzzRows(
+      memory::MemoryPool* pool,
+      size_t rowsNum,
+      const RowTypePtr& rowType) {
+    VectorFuzzer fuzzer({.vectorSize = rowsNum}, pool);
+    VectorFuzzer fuzzerWithNulls(
+        {.vectorSize = rowsNum, .nullRatio = 0.7}, pool);
+    if (rowType->children().size() == 1) {
+      return fuzzer.fuzzRow(rowType);
+    }
+    std::vector<VectorPtr> children;
+    for (int column = 0; column < rowType->children().size() - 1; ++column) {
```

Comment:
Why not use fuzzer.fuzzInputRow(rowType)?

Reply:
we use two fuzz instance here to enforce pull path compare

Reply:
@skadilover I'm not sure I understand. What does "pull path compare" mean? Would you clarify?

Reply:
For example, for col1, col 2, col3, and order by col1, col2, col3
1. If we random generate all column ,  maybe we only compare col1 when run benchmark ,because it has high selectivty, 
2. I use a fuzzerWithNulls wih 0.7 null rait to generate col1 and col2, so that 70% data of them are same,  by this way, we can mock a real multi-case